apps/pearl-api/pages/web3auth/swap-owner-session.tsx

```diff
@@ -0,0 +1,371 @@
+import { CheckCircleOutlined, CloseCircleOutlined, LoadingOutlined } from '@ant-design/icons';
+import Safe from '@safe-global/protocol-kit';
+import { useWeb3Auth, useWeb3AuthConnect } from '@web3auth/modal/react';
+import { Alert, Card, Flex, Space, Spin, Typography } from 'antd';
+import { useRouter } from 'next/router';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { createGlobalStyle } from 'styled-components';
+import { Address } from 'viem';
+
+import { Web3AuthProvider } from 'context/Web3AuthProvider';
+
+import { EvmChainId, EvmChainName, toHexChainId } from '../../utils';
+
+const { Title, Paragraph, Text } = Typography;
+
+export const Styles = createGlobalStyle`
+  .w3a-parent-container > div {
+    background-color: rgba(0, 0, 0, 0.45) !important;
+  }
+`;
+
+enum Events {
+  WEB3AUTH_SWAP_OWNER_MODAL_INITIALIZED = 'WEB3AUTH_SWAP_OWNER_MODAL_INITIALIZED',
+  WEB3AUTH_SWAP_OWNER_RESULT = 'WEB3AUTH_SWAP_OWNER_RESULT',
```

Comment:
I'd rather split into WEB3AUTH_SWAP_OWNER_SUCCESS/WEB3AUTH_SWAP_OWNER_ERROR
so that it's clear what data to expect on the other side (e.g. txn for success, errorMessage for fail)